flask_security/cache.py

```diff
@@ -0,0 +1,38 @@
+from hashlib import md5
+
+from flask import current_app
+from cachetools import TTLCache
+
+
+class VerifyHashCache:
+    """Cache handler to make it quick password check by bypassing
+    already checked passwords against exact same couple of token/password.
+    This cache handler is more efficient on small apps that
+    run on few processes as cache is only shared between threads."""
+
+    def __init__(self):
+        ttl = current_app.config.get("VERIFY_HASH_CACHE_TTL", 60 * 5)
+        max_size = current_app.config.get("VERIFY_HASH_CACHE_MAX_SIZE", 500)
+        self._cache = TTLCache(max_size, ttl)
+
+    def get_hash(self, token, user):
+        """Generate a md5 hash for unique key for given token
+        and user password, so the md5 is quick to compute, it
+        hashes content, and should be enough to avoid collisions."""
+        key = "{} {}".format(user.password, token).encode()
+        return md5(key).hexdigest()
+
+    def has_verify_hash_cache(self, token, user):
+        """Check given token/password verification is in cache."""
+        key_hash = self.get_hash(token, user)
+        cache = self._cache.get(key_hash)
+        return cache
+
+    def set_cache(self, token, user):
+        """When a password is checked, then result is put in cache."""
+        key_hash = self.get_hash(token, user)
```

Comment:
While I think this works - I would be more inclined to use user.id (which is guaranteed to be unique) as the key and store the token as the value - then you can just compare, and no need for md5 or anything..

Reply:
ok for that, I'll do it.